### PR TITLE
Add new test cases for restore package

### PR DESCRIPTION
**Pull request by CiciLi1**
# Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/3108

## Description
Add the below two new test cases about the "Restore Package" feature to the NuGet.Client-VSDaily Tests pipeline.

1. [Verify the deleted "project.assets.json" file can be restored with context menu “Restore NuGet Packages” in a non-SDK style project](https://microsoft.sharepoint.com/teams/NuGet/_layouts/OneNote.aspx?id=%2Fteams%2FNuGet%2FTeam%2FManual%20tests%2FTest%20Cases&wd=target%28Daily%20Manual%20Test.one%7C5CCE9D00-EFB1-411C-A3C5-268C164F5FE3%2FRestore%20Package%20%284%5C%29%7C8CC67692-3ED9-4205-B855-E6056C3BD099%2F%29)
2. [Verify the deleted "project.assets.json" file can be restored with auto-restore when reloading an SDK style project](https://microsoft.sharepoint.com/teams/NuGet/_layouts/OneNote.aspx?id=%2Fteams%2FNuGet%2FTeam%2FManual%20tests%2FTest%20Cases&wd=target%28Daily%20Manual%20Test.one%7C5CCE9D00-EFB1-411C-A3C5-268C164F5FE3%2FRestore%20Package%20%284%5C%29%7C8CC67692-3ED9-4205-B855-E6056C3BD099%2F%29)
## PR Checklist

- [X] Meaningful title, helpful description and a linked NuGet/Home issue
- [X] Added tests
- [X] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

File: test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs [hunk 1]
```diff
@@ -877,6 +877,46 @@ public async Task UninstallPackageFromFSharpFromUI()
...
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+
+            var assetsFilePath = CommonUtility.GetAssetsFilePath(project.FullPath);
+            CommonUtility.WaitForFileExists(new FileInfo(assetsFilePath));
+            File.Delete(assetsFilePath);
+
+            // Act
+            if (projectTemplate.Equals(ProjectTemplate.ConsoleApplication))
+            {
+                CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
```
Comment (jeffkl) [inline test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Just curious, why is there different behavior for different project types?

**Review by jeffkl — DISMISSED** [review]

File: test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs (hunk 1 above)
Comment (nkolev92) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Legacy PR vs SDK PR which has auto restore I assume?

If my assumption is correct, why do we need `             CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);` @CiciLi1?

uiwindow.InstallPackageFromUI should have created a project.assets.json.

Comment (CiciLi1) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Yes, SDK PR has auto restore feature. we deleted the project.assets.json file before restoring package.
(The test case is to verify that the deleted “project.assets.json” file can be restored by the restore feature)

Comment (nkolev92) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Automatic restore should not be triggered by deleting an assets file, so when would line 913 become relevant?

**Review by jgonz120 — DISMISSED** [review]
lgtm

File: test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs (hunk 1 above)
Comment (CiciLi1) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Thanks a lot for the reminder, we thought the `WaitForAutoRestore()` contains restoring project step. And now we have added "unload and reload projects" to the SDK project to trigger the auto-restore feature. Could you please help to review it?

Comment (nkolev92) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Is there a specific reason why the test case needs a reload?
Restore triggered through:

```cs
                CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
```

should do the job.

Comment (CiciLi1) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: We previously had test cases that triggered the package restoring through `CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);`. The purpose of the new test case is to verify that the package file can be restored through automatic restoration. so we use the "unload and reload projects" to trigger the auto-restore.

Comment (nkolev92) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: If the intent of this test is to verify unload/reload handles assets file regeneration, then I think the test should be calling that out in the name.

This is a scenario where I think we'd rather see 2 different test cases. 

You can share set-up code through a tertiary method.

Comment (CiciLi1) [reply test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs]: Thanks to your suggestion, I have split the test cases into two different scenarios, adjusted the names of the test cases. we wrote a  tertiary method for set-up code(unload and reload project).

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**